device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -242,7 +242,7 @@ public void onMessageSent(Message message, String deviceId, Throwable e)
             try
             {
                 String messageId = message.getCorrelationId();
-                if (messageId != null && correlationCallbacks.containsKey(messageId))
```

Comment:
Why remove `messageId != null`? If null an exception would be thrown.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.containskey?view=net-5.0#exceptions

Reply:
Actually yes, you're right. I'm using a concurrent hashmap so supplying a null will throw an NPE.